cloudera-on-centos/azuredeploy.json

```diff
@@ -1,314 +1,407 @@
-            "type": "string",
-            "defaultValue": "AzureRM-Keys",
-            "metadata": { 
-                "description": "The name of the key vault which provides the private key used for SSH login.  See the readme for details." 
-            } 
+  "$schema": "http://schema.management.azure.com/schemas/2014-04-01-preview/deploymentTemplate.json",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "adminUserName": {
+      "type": "string",
+      "defaultValue": "testuser",
+      "metadata": {
+        "description": "Admin user name for the Virtual Machines"
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "defaultValue": "Eur32#1e",
+      "metadata": {
+        "description": "Admin password for the Virtual Machines"
+      }
+    },
+    "cmUsername": {
+      "type": "string",
+      "defaultValue": "cmadmin",
+      "metadata": {
+        "description": "User name for the Cloudera Manager"
+      }
+    },
+    "cmPassword": {
+      "type": "securestring",
+      "defaultValue": "cmpassword",
+      "metadata": {
+        "description": "password for the Cloudera Manager"
+      }
+    },
+    "storageAccountPrefix": {
+      "type": "string",
+      "defaultValue": "storageaccountprefix",
+      "metadata": {
+        "description": "Unique namespace for the Storage Account where the Virtual Machine's disks will be placed"
+      }
+    },
+    "dnsNamePrefix": {
+      "type": "string",
+      "defaultValue": "dnsnameprefix",
+      "metadata": {
+        "description": "Unique public dns name where the Virtual Machines will be exposed"
+      }
+    },
+    "region": {
+      "type": "string",
+      "defaultValue": "japanwest",
+      "metadata": {
+        "description": "Location where resources will be provisioned"
+      }
+    },
+    "storageAccountType": {
+      "type": "string",
+      "defaultValue": "Premium_LRS",
+      "metadata": {
+        "description": "The type of the Storage Account to be created (Default to Standard_LRS)"
+      }
+    },
+    "virtualNetworkName": {
+      "type": "string",
+      "defaultValue": "clouderavnet",
+      "metadata": {
+        "description": "The name of the virtual network provisioned for the deployment"
+      }
+    },
+    "subnetName": {
+      "type": "string",
+      "defaultValue": "clouderasubnet",
+      "metadata": {
+        "description": "Subnet name for the virtual network where resources will be provisioned"
+      }
+    },
+    "addressPrefix": {
+      "type": "string",
+      "defaultValue": "10.0.0.0/24",
+      "metadata": {
+        "description": "The network address space for the virtual network"
+      }
+    },
+    "subnetPrefix": {
+      "type": "string",
+      "defaultValue": "10.0.0.0/24",
+      "metadata": {
+        "description": "The network address space for the virtual subnet"
+      }
+    },
+    "nodeAddressPrefix": {
+      "type": "string",
+      "defaultValue": "10.0.0.",
```

Comment:
Consider making this a variable to reduce the number of parameters a user has to enter unless the user absolutely needs to know the node address prefix